### DISCUSSION TITLE
No cluster fork support in Deno v2?

### DISCUSSION THREAD

**Discussion by melroy89** [denoland/deno#26456] [Q&A] [answered]
Deno v2 [promised backwards compatibility with Node.js](https://deno.com/blog/v2.0).. So I tried Deno v2 again after 3 years... 

1. I needed to fix some imports with `node:` prefix
2. Since I'm using typescript it's using `.js` as imports, so the workaround for now is using: `--unstable-sloppy-imports`
3. Most importantly now: there is no cluster support!? While you are promoting yourself as Node.js compatible with Deno v2.

And I got:

```sh
Master 73828 is running
error: Uncaught (in promise) Error: Not implemented: cluster.fork
    at notImplemented (ext:deno_node/_utils.ts:9:9)
    at EventEmitter.fork (node:cluster:18:3)
```

What a shame. 😢 Is Deno not supporting Node.js cluster??

```sh
deno --version

deno 2.0.2 (stable, release, x86_64-unknown-linux-gnu)
v8 12.9.202.13-rusty
typescript 5.6.2
```

Regards,
Melroy

**Comment by rmcarias** [comment]
I am facing a similar problem. We use a lib called 'throng' and it uses node cluster. I hope they add this compatibility. Or a workaround.

**Comment by bartlomieju** [comment, marked as answer]
`node:cluster` is supported as of **Deno 2.8.0**. `cluster.fork()` and the primary/worker lifecycle were implemented in #33752 (shipped in 2.8.0), so the `Not implemented: cluster.fork` error you hit on 2.0.2 no longer occurs. Libraries built on `node:cluster` (e.g. `throng`, cc @rmcarias) work now.

Please `deno upgrade` to 2.8.0 or newer and give it another try. One caveat: the implementation currently targets Unix; Windows support is still limited. If you run into a specific gap, please open a new issue with a repro.
